Sources/idt/idt.cc

```diff
@@ -89,6 +89,20 @@ class visitor : public clang::RecursiveASTVisitor<visitor> {
     return context_.getFullLoc(TD->getBeginLoc()).getExpansionLoc();
   }
 
+  template <typename Decl_>
+  bool decl_in_header(const Decl_ *D) const {
```

Comment:
What do you think of naming this `is_in_header`? I don't think that the `decl_` prefix helps here. We could further overload this in the future for different types if desired to get a similar check for say `GlobalValue`.